docs/devguide.rst

```diff
@@ -55,39 +57,77 @@ Submitting a PR
 Once you are done making changes, you have to submit your contribution through a pull request (PR).
 The procedure for submitting a PR is the following.
 
-1. Make sure all tests still pass:
+1. Make sure all tests still pass, the code is free of lint, and you docstrings compile correctly:
 
    .. code-block:: bash
 
+        invoke lint
         invoke test
+        invoke docs
 
 2. Add yourself to ``AUTHORS.md``.
 3. Commit your changes and push your branch to GitHub.
-4. Create a `pull request <https://help.github.com/articles/about-pull-requests/>`_ through the GitHub website.
-
-During development, use `pyinvoke <http://docs.pyinvoke.org/>`_ tasks on the
-command line to ease recurring operations:
-
-* ``invoke clean``: Clean all generated artifacts.
-* ``invoke check``: Run various code and documentation style checks.
-* ``invoke docs``: Generate documentation.
-* ``invoke test``: Run all tests and checks in one swift command.
+4. Create a `pull request <https://help.github.com/articles/about-pull-requests/>`_.
 
 
 Style guide
 ===========
 
-* PEP 8
-* flake8
-* naming conventions
-* consistency
-* foolish consistency
-* principle of least astonishment
+The command ``invoke lint`` runs the entire codebase through ``flake8``.
+As described in the `docs <https://flake8.pycqa.org/en/latest/manpage.html>`_,
+``flake8`` includes lint checks provided by the PyFlakes project,
+PEP-0008 inspired style checks provided by the PyCodeStyle project,
+and McCabe complexity checking provided by the McCabe project.
+
+The list of potential error codes issued by ``flake8`` are available here:
+https://flake8.pycqa.org/en/latest/user/error-codes.html
+
+The PEP-0008 style guide is available here:
+https://www.python.org/dev/peps/pep-0008/
+
+Note that the maximum line length is set to 180 rather 79 in the ``setup.cfg`` of the repo.
+
+
+Naming conventions
+==================
+
+We (intend to) use the following naming conventions.
+
+1. variables, functions, methods, attributes use "snake_case":
+   they are written in lowercase and spaces between words are replaced by underscores.
+
+2. class names use (Upper) "CamelCase":
+   The are written in lowercase, with the first letter of each word capitalized
+   and spaces between words removed.
+
+3. module or package level variables are in uppercase
+   and with spaces between words replaced by underscores.
+
+
+Quotes
+======
+
+Ideally, we would use the following conventions for quotes.
+
+1. Double quotes for multiline statements (``"""``).
```

Comment:
This might be a little confusing for the uninitiated I think.  (``"""``? But there are three of them? What do they mean double quotes?) So perhaps either write "Double quotation marks" or use an example  of the form (``"""Calculates something useful."""``).